### ISSUE TITLE
Feature request: add --external-chat switch to allow Aider to receive individual messages written in a custom text editor

### ISSUE THREAD

**Issue by insilications** [paul-gauthier/aider#1818]
### Issue

# Description

The `--external-chat <text_editor_path>` switch would allow Aider to receive individual message written in a custom text editor.

1. The `--external-chat` switch alone would imply reading the path to the text editor from the `AIDER_CHAT_EDITOR` environment variable.
2. When typing `/architect`, `/ask` or `/code` without any message attached to it, Aider opens a temporary text file in the message editor that the user has selected as their preferred message editor.
3. Once this is done, Aider waits for the user to type a custom message in the message editor, save it to the aforementioned temporary text file, and finally close the editor. When Aider detects that the user has closed the editor process, it reads the content of the temporary text file and uses it to send the chat message content.
4. Aider will try to perform all necessary cleanup steps, such as deleting the temporary text file.
5. The only requirement for the program to work properly is that the chosen message editor must open the temporary file in a blocking mode, which allows us to detect when the process of the editor that opened the temporary file is closed.

Small python example. Set `AIDER_CHAT_EDITOR` environment variable to the chosen text editor:

```python
import os
import subprocess
import tempfile
import shlex
import shutil
from pathlib import Path

def get_editor_input():
    msg_editor = os.environ.get('AIDER_CHAT_EDITOR')
    if not msg_editor:
        print("No message editor set to AIDER_CHAT_EDITOR.")
        return None

    msg_editor_cmd = shlex.split(msg_editor)
    editor_executable = shutil.which(msg_editor_cmd[0])
    if not editor_executable:
        print(f"Error: {msg_editor_cmd[0]} executable not found.")
        return None

    try:
        with tempfile.NamedTemporaryFile(mode='w+', suffix=".txt", delete=False, encoding='utf-8') as temp_file:
            temp_filename = temp_file.name

        temp_path = Path(temp_filename)
        result = subprocess.run(msg_editor_cmd + [str(temp_path)], check=True)

        if result.returncode == 0:
            return temp_path.read_text().strip()
        else:
            print(f"Error: {editor_executable} exited with a non-zero status.")
            return None

    except subprocess.CalledProcessError as e:
        print(f"Error: {editor_executable} exited with a non-zero status. {e}")
    except FileNotFoundError:
        print(f"Error: {editor_executable} executable not found.")
    except Exception as e:
        print(f"An unexpected error occurred: {e}")
    except KeyboardInterrupt:
        print("Operation cancelled by user.")
    finally:
        if temp_path.exists():
            temp_path.unlink(missing_ok=True)

    return None

if __name__ == "__main__":
    commit_message = get_editor_input()

    if commit_message:
        print(f"Your custom message: {commit_message}")
    else:
        print("No custom message provided or an error occurred.")
```

# Use case

While Aider's terminal interface is certainly nice, it might be beneficial to consider allowing users to type chat messages in their external text editors of choice.

**Comment by fry69** [comment]
Related or duplicate of #1315

**Comment by insilications** [comment]
> Thank you for filing this issue.
> 
> This seems to be related or duplicate feature request of #1315. Since I cannot see a major difference to the `/editor` approach from #1315 I tend to close this issue as a duplicate.
> 
> Please respond if you have a different opinion and/or I have missed something.

Yes, I will close it.
